lib/manager/pre-commit/util.ts

```diff
@@ -0,0 +1,36 @@
+export type PrecommitGitDependency = {
+  repo: string;
+  rev: string;
+};
+
+/**
+ * This is a workaround helper to allow the usage of 'unknown' in
+ * a type-guard function while checking that keys exist.
+ *
+ * @see https://github.com/microsoft/TypeScript/issues/21732
+ * @see https://stackoverflow.com/a/58630274
+ */
+export function hasKey<K extends string, T>(
```

Comment:
@viceice are you ok with this, and if so then should it be moved to a shared location?

Reply:
Yes, I think we do have such a function at another place. Time to move it to utils.

Reply:
Let's keep it here for this PR as we don't have an obvious existing location for it except a new file

Reply:
I moved it to `util/object.ts`, latest push also contains a corresponding spec.